src/components/pages/users-page/users-page.tsx

```diff
@@ -10,129 +12,326 @@ import {
   TableRow,
   DialogActions,
   Dialog,
-  DialogTitle,
   DialogContent,
   DialogContentText,
-  IconButton, FormControl, InputLabel, Select,
+  IconButton,
+  FormControl,
+  InputLabel,
+  Select,
 } from '@material-ui/core';
 import React, { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import MailOutlineIcon from '@material-ui/icons/MailOutline';
 import CheckIcon from '@material-ui/icons/Check';
-import axios from 'axios';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+import ReportProblemOutlinedIcon from '@material-ui/icons/ReportProblemOutlined';
+import axios, { AxiosError } from 'axios';
+import axiosRetry from 'axios-retry';
 import useStyles from './users-page.styles';
 import { UserState } from '../../../reducers/user.reducer';
 import { AppState } from '../../../store';
 import { ApiRole, ApiUser, ApiAccessRequest } from '../../../models/api-response';
 import { AppFrame } from '../../../actions/app-frame.actions';
 import { ButtonWithSpinner } from '../../buttons/button-with-spinner';
+import { AlertDialog, AlertDialogProps } from '../../alert-dialog/alert-dialog';
+
+axiosRetry(axios, {
+  retries: 3,
+  retryDelay: axiosRetry.exponentialDelay,
+});
 
 interface AccessRequestRow extends ApiAccessRequest {
   waiting?: boolean,
 }
 
+type UserMoreMenuState = {
+  element: HTMLElement
+  loading: boolean
+  showChangeUserRole: Boolean
+  showRemoveUserFromGroup: Boolean
+  user: ApiUser
+};
+
+type RoleSelectionDialogProps = {
+  availableRoles: ApiRole[]
+  confirmButtonText: string
+  loading: boolean
+  onCancel?: () => void
+  onChange: (selectedRole: ApiRole) => void
+  open?: boolean
+  user?: ApiUser
+};
+
+const RoleSelectionDialog: React.FunctionComponent<RoleSelectionDialogProps> = ({
```

Comment:
Can you move this dialog to a separate file?  We have been doing that for all of the other pages and it would be nice to keep things consistent.

Reply:
Yeah... sorry the laziness here. :/

I knew this was going to get another refactor with the "assign role dialog" changes so I wasn't too worried about it being in the same file. It's already pulled out in the "assign role dialog" branch that I'll be pushing in a few.